code/modules/client/preferences.dm

```diff
@@ -1976,3 +1982,24 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 			return
 		else
 			custom_names[name_id] = sanitized_name
+
+/// Handles adding and removing donator items from clients
+/datum/preferences/proc/handle_donator_items(client/user)
+	parent = user
+	var/req_donator = CONFIG_GET(flag/donator_items)
+	var/datum/loadout_category/DLC = GLOB.loadout_categories["Donator"] // stands for donator loadout category but the other def for DLC works too xD
+
+	if(IS_PATRON(user.ckey) || !req_donator)
+		for(var/key in DLC.gear)
+			var/datum/gear/donator/AG = GLOB.gear_datums[key]
+			if(!(AG.id in purchased_gear))
+				purchased_gear += AG.id
+				AG.purchase(user)
+		save_preferences()
+	else
+		if(purchased_gear.len || equipped_gear.len)
+			for(var/key in DLC.gear)
+				var/datum/gear/donator/RG = GLOB.gear_datums[key]
+				equipped_gear -= RG.id
+				purchased_gear -= RG.id
+			save_preferences()
```

Comment:
This might have the unintended side effect of saving preferences set other places that were not intended to be saved

Reply:
That's how everything else in gear code does it, I'll be working on a gear rework PR soon. I'll make sure to address that too xd